### DISCUSSION TITLE
Migration to version 7 - useNavigation problem

### DISCUSSION THREAD

**Discussion by Paul6552** [remix-run/react-router#12484] [Q&A]
I am doing the migration from version 6 and 7. So far so good, but the migration guide says that I can remove `react-router-dom`.
But I have one function `useNavigation` which relies on that. I know `react-router `has the function `Navigation`, but this works different than `useNavigation`.

Is there any other replacement for this function or any other way how to do it know?

**Comment by khabubuphathu** [comment]
The upgrade guide says replace `react-router-dom` with `react-router`. You should be able to import the `useNavigation` hook as follows
`import { useNavigation } from "react-router"`

**Reply by Paul6552** [reply]
And I am not able to import useNavigation from react-router. There is not such a function.

**Comment by Paul6552** [comment]
I am stupid. I dont know why but now it works. Thanks and sry for the noise
